lib/payment_processor/clients/braintree/subscription.rb

```diff
@@ -6,6 +6,13 @@ def self.make_subscription(amount:, plan_id:, payment_method_token:)
           new(amount, plan_id, payment_method_token).subscribe
         end
 
+        def self.make_subscription_from_transaction(transaction)
+          amount = transaction...
+          plan_id = ENV['BRAINTREE_SUBSCRIPTION_PLAN_ID']
+          payment_method_token = transaction...
+          new(amount, plan_id, payment_method_token).subscribe
+        end
+
```

Comment:
I've implemented this approach on my branch. One of the issues is that in the controller, this returns one success object. First there's a success object for the transaction itself, which doesn't get passed on to the controller - then there's a success object for the subscription, which gets returned as `{ success: true, transaction_id: <subscription_id> }` because it's a subscription passed through the transaction method. The spec I wrote in `braintree_controller_spec.rb` is written as if this was desired behavior and will need to be modified if we want to handle this some other way.